orangecontrib/text/widgets/owscraper.py

```diff
@@ -0,0 +1,215 @@
+
+    def repopulate_url_combo(self, data):
+        """
+        change in attribute selected
+        :param data:
+        :return: attribute name
+        """
+        from itertools import chain
+        self.metas = [a for a in chain(data.domain.metas,
+                                       data.domain.attributes,
+                                       data.domain.class_vars)
+                      # Filter string variables
+                      if (a.is_discrete and a.values and isinstance(a.values[0], str) and not a.ordered or
+                          a.is_string)] if data else []
+        self.url_combo.clear()
+        self.selected_url = 0
+        for i, var in enumerate(self.metas):
+            self.url_combo.addItem(gui.attributeIconDict[var], var.name)
+            # Select default url attribute
+            if var.name.lower() == 'url':
+                self.selected_url = i
+        if self.metas:
+            self.url_combo.setCurrentIndex(self.url_combo.findText(self.metas[self.selected_url].name))
+        return self.metas[self.selected_url].name
+
+    def get_list(self, data, colname):
+        """
+        returns list of urls in the given column
+        :param data:
+        :param colname:
+        :return: list
+        """
+        li = data.metas[:,0] #default
+
+        for i in range(len(data.domain.metas)):
+            if colname==data.domain.metas[i].name:
+                li=data.metas[:,i]
+        return li
+
+    def generate_table(self, url_list):
+        """
+        generate new table with values requested by user
+        :param url_list:
+        :return: table
+        """
+        new_table=None
+        text_includes_params = [self.includes_article, self.includes_author, self.includes_date, 
+                                 self.includes_title, self.includes_web_url]
+        required_text_fields = [incl_field for yes, incl_field in zip(text_includes_params, ARTICLE_TEXT_FIELDS) if yes]
+        meta_vars = [StringVariable.make(field) for field in required_text_fields]
+        metadata=[]
+        for url in url_list:
+            info=self.get_info(url)
+            final_fields = [incl_field for yes, incl_field in zip(text_includes_params, info) if yes]
+            metadata.append(final_fields)
+        metadata = np.array(metadata, dtype=object)
+        metas=metadata
+        domain = Domain([], class_vars=None, metas=(meta_vars))
+        new_table = Corpus(None, None, metadata, domain, meta_vars)
+        self.send("Data",new_table)
+
+    def set_data(self, data):
+        """
+        gets the input data
+        generates a new table with the requested values
+        :param : data
+        """
+        self.data = data
+        self.colname=self.repopulate_url_combo(data)
+        self.url_list=self.get_list(data,self.colname)
+        self.generate_table(self.url_list)
+
+def main():
+    app = QApplication([])
```

Comment:
`unresolved reference to QApplication`. Can you please fix? To test, try running this file directly, not through Orange.